hal/src/mrc/include/mrc/NetComm.h

```diff
@@ -16,56 +18,94 @@
-    return OpModeTrace{
-        .Hash = Value & 0x3FFFFFFFFFFFFFFF,
-        .Enabled = (Value & 0x8000000000000000) == 0,
-        .NoCurrentOpMpode = (Value & 0x4000000000000000) != 0,
-    };
+struct OpModeHash {
+  uint64_t Hash : 56 {};
+  uint64_t IsAuto : 1 {};
+  uint64_t IsTest : 1 {};
+  uint64_t IsEnabled : 1 {};
+  uint64_t Reserved : 6 {};
+
+  static OpModeHash MakeTest(uint64_t Hash, bool Enabled = false) {
+    OpModeHash FullHash;
+    FullHash.Hash = Hash & 0x00FFFFFFFFFFFFFF;
+    FullHash.IsTest = 1;
+    FullHash.IsEnabled = Enabled ? 1 : 0;
+    return FullHash;
+  }
+
+  static OpModeHash MakeTele(uint64_t Hash, bool Enabled = false) {
+    OpModeHash FullHash;
+    FullHash.Hash = Hash & 0x00FFFFFFFFFFFFFF;
+    FullHash.IsEnabled = Enabled ? 1 : 0;
+    return FullHash;
+  }
+
+  static OpModeHash MakeAuto(uint64_t Hash, bool Enabled = false) {
+    OpModeHash FullHash;
+    FullHash.Hash = Hash & 0x00FFFFFFFFFFFFFF;
+    FullHash.IsAuto = 1;
+    FullHash.IsEnabled = Enabled ? 1 : 0;
+    return FullHash;
+  }
+
+  static OpModeHash FromValue(uint64_t Value) {
+    OpModeHash RetVal;
+    RetVal.Hash = Value & 0x00FFFFFFFFFFFFFF;
+    RetVal.IsAuto = (Value & 0x2000000000000000) != 0;
+    RetVal.IsTest = (Value & 0x4000000000000000) != 0;
+    RetVal.IsEnabled = (Value & 0x8000000000000000) != 0;
+    return RetVal;
   }
 
   uint64_t ToValue() const {
-    uint64_t RetVal = Hash & 0x3FFFFFFFFFFFFFFF;
-    RetVal |= Enabled ? 0 : 0x8000000000000000;
-    RetVal |= NoCurrentOpMpode ? 0x4000000000000000 : 0;
+    uint64_t RetVal = Hash & 0x00FFFFFFFFFFFFFF;
+    RetVal |= IsAuto ? 0x2000000000000000 : 0;
+    RetVal |= IsTest ? 0x4000000000000000 : 0;
+    RetVal |= IsEnabled ? 0x8000000000000000 : 0;
     return RetVal;
   }
 };
 
 struct ControlFlags {
-  uint32_t Enabled : 1;
-  uint32_t Auto : 1;
-  uint32_t Test : 1;
-  uint32_t EStop : 1;
-  uint32_t FmsConnected : 1;
-  uint32_t DsConnected : 1;
-  uint32_t WatchdogActive : 1;
-  uint32_t Alliance : 6;
-  uint32_t Reserved : 19;
+  uint32_t Enabled : 1 {};
+  uint32_t Auto : 1 {};
+  uint32_t Test : 1 {};
+  uint32_t EStop : 1 {};
+  uint32_t FmsConnected : 1 {};
+  uint32_t DsConnected : 1 {};
+  uint32_t WatchdogActive : 1 {};
+  uint32_t Alliance : 6 {};
+  uint32_t Reserved : 19 {};
 };
 
 struct JoystickAxes {
  public:
-  std::span<int16_t> Axes() { return std::span{AxesStore.data(), GetCount()}; }
+  uint16_t AvailableAxes{0};
+
+  std::span<int16_t> Axes() {
+    return std::span{AxesStore.data(),
+                     static_cast<size_t>(GetMaxAvailableCount())};
+  }
 
   std::span<const int16_t> Axes() const {
-    return std::span{AxesStore.data(), GetCount()};
+    return std::span{AxesStore.data(),
+                     static_cast<size_t>(GetMaxAvailableCount())};
   }
 
-  void SetCount(uint8_t NewCount) {
-    Count = (std::min)(NewCount, static_cast<uint8_t>(MRC_MAX_NUM_AXES));
+  void SetMaxAvailableToCount(size_t Count) {
```

Comment:
```suggestion
  void SetMaxAvailableCount(size_t Count) {
```
Felt inconsistent with `GetMaxAvailableCount()`, might just be me though.